src/editors/object.js

```diff
@@ -698,7 +698,7 @@ JSONEditor.defaults.editors.object = JSONEditor.AbstractEditor.extend({
     if(this.jsoneditor.options.remove_empty_properties || this.options.remove_empty_properties) {
       for(var i in result) {
         if(result.hasOwnProperty(i)) {
-          if(typeof result[i] === 'undefined' || result[i] === '') delete result[i];
+          if(typeof result[i] === 'undefined' || result[i] === '' || Object.keys(result[i]).length == 0 && result[i].constructor == Object) delete result[i];
```

Comment:
Wrong logic. It's should be...
`if(typeof result[i] === 'undefined' || result[i] === ''  || (Object.keys(result[i]).length == 0 && result[i].constructor == Object)) delete result[i];`